domain/src/main/kotlin/feature/BackupManager.kt

```diff
@@ -0,0 +1,62 @@
+package ltd.evilcorp.domain.feature
+
+import android.content.res.Resources
+import android.net.Uri
+import androidx.core.os.ConfigurationCompat
+import com.fasterxml.jackson.module.kotlin.jacksonObjectMapper
+import ltd.evilcorp.core.repository.MessageRepository
+import ltd.evilcorp.core.vo.Message
+import java.text.DateFormat
+import java.text.SimpleDateFormat
+import java.util.*
+import javax.inject.Inject
+import javax.inject.Singleton
+import kotlin.collections.List
+import kotlinx.coroutines.flow.first
+import kotlinx.coroutines.flow.single
+import kotlinx.coroutines.runBlocking
+
+@Singleton
+class BackupManager @Inject constructor(
```

Comment:
Let's call this feature "Export" instead of "Backup". It's what we're using in other places, and it goes nicer with "Export/Import" once we support importing from this data as well.